File: lclang/src/main/java/lclang/statements/expressions/StopExpression.java
package lclang.statements.expressions;

import lclang.Caller;
import lclang.LCBaseExecutor;
import lclang.Link;
import lclang.libs.lang.classes.VoidClass;

public class StopExpression extends Expression {
    public static final Link STOP_VALUE = new Link(VoidClass.instance, Link.State.STOP);

    public StopExpression(int line) {
        super(line);
    }

    @Override
    public Link visit(Caller prevCaller, LCBaseExecutor visitor) {
        return STOP_VALUE;
    }
}

File: lclang/src/main/kotlin/lclang/statements/StopExpression.kt
package lclang.statements

import lclang.Caller
import lclang.LCBaseExecutor
import lclang.Value
import lclang.types.Types

class StopExpression(
    line: Int,
    column: Int
) : Expression(line, column) {
    override fun visit(prevCaller: Caller, visitor: LCBaseExecutor) = Value(Types.VOID, null, Value.State.STOP)
}